src/openrct2/paint/Paint.cpp

```diff
@@ -477,15 +477,18 @@ template<int TRotation> static void PaintSessionArrangeImpl(PaintSessionCore& se
         return;
     }
 
-    PaintStructsLinkQuadrants(session);
+    PaintStruct psHead{};
```

Comment:
i think we should add some comments on this. My understanding is as follows:

The psHead is a temporary that never has any information set to it it just acts as something for things to attach on to. After we have setup everything we set the psHead to the first real paintstruct (or nullptr if there is none?).